### ISSUE TITLE
Installer - Desktop Bridge

### ISSUE THREAD

**Issue by karlgjertsen** [paolosalvatori/ServiceBusExplorer#49]
None of the authors links to installers work. Where can I get an installer from?

**Comment by janpieterz** [comment]
As far as I know there is no installer that is officially built by the authors (and signed etc etc).

You'll have to build it yourself it you want one currently!

**Comment by paolosalvatori** [comment]
Hi guys
In the past I tried to make the tool on chocolatey but I didn't have enough time to complete the task. If any of you wants to help me to build an installer. I'll be on vacation in August, but ping me afterwards if you want to collaborate.
Ciao
Paolo

**Comment by janpieterz** [comment]
@paolosalvatori more than happy to lend a hand! For my own business we regularly use 'simple' WiX installers, if that is of any help here. Chocolatey is a good place to be for a tool like this too.

Have a nice vacation, I'll pop back in here in September and see what we can start!

**Comment by paolosalvatori** [comment]
Thanks Jan, see you in September :)

**Comment by msimecek** [comment]
Hi, how about publishing it to the Windows Store as a package converted by the [Desktop App Converter](https://developer.microsoft.com/en-us/windows/bridges/desktop)? That would make getting SBE much easier and it's supported by WiX ;)

**Comment by paolosalvatori** [comment]
Yes. This sounds like a great idea :) Do we have an easy way to regenerate a new version any time a new version of SBE is available?

**Comment by SeanFeldman** [comment]
If I may suggest the following:
1. Establish versioning on the master using Git(Hub)Flow (raised an issue in the past #29 )
2. Establish CI/CD (re-opened #2) following the suggestion by @regisbsb 

Combined together, we'll be able to get versioned releases deployed via GH. 
Down the road, we'll be able to replace that with an installer (if needed).

**Comment by msimecek** [comment]
@paolosalvatori I will try to create an APPX package and an update mechanism for it.

**Comment by paolosalvatori** [comment]
Thanks @msimecek!

**Comment by SeanFeldman** [comment]
@msimecek any updates on this?

**Comment by msimecek** [comment]
@SeanFeldman Yes, I prepared a build process in Visual Studio Team Services which takes sources from this repo on GitHub and builds an APPX package. I will now get in touch with @paolosalvatori to make it "official" and published to Windows Store.

**Comment by SeanFeldman** [comment]
Any chance we could get it going with AppVeyor which we started using?

**Comment by paolosalvatori** [comment]
Hi @msimecek thanks for your help. Just let me know what I should do to build the package and push on it on the Windows Store, thanks! @SeanFeldman, let me know how I can use AppVeyor.

**Comment by SeanFeldman** [comment]
@paolosalvatori you have access to https://ci.appveyor.com (`ServiceBusExplorer` project).
Please DM if you can't access it. Used your private email for invite a month ago.

**Comment by paolosalvatori** [comment]
Hi @SeanFeldman I verified the access, let me know if I should anything.

**Comment by SeanFeldman** [comment]
> if I should anything.

@paolosalvatori my mind reading is not that good 😺 What did you want to ask?

Can we have a PR that for this feature? That way we can collaboratively make it work.
As of today, `master` is deploying a zip with SBE to GH releases section. We could integrate work done here to also have APPX either deployed to GH releases as well, or rather deploy to Windows Store (sorry, not familiar with how it works).

@msimecek could you start the PR? Any secrets should **not** be stored as-is in appveyor.yml, but rather encoded and stored on AppVeyor server, which both @paolosalvatori and myself can help with.

Let me know if this plan of attack works. Thanks.

**Comment by SeanFeldman** [comment]
Any update on this @msimecek @paolosalvatori ?

**Comment by paolosalvatori** [comment]
Not from me guys :)

**Comment by SeanFeldman** [comment]
@msimecek are you still committed to this? If you need assistance with transitioning from biuld process from VSTS to AppVeyor, I can help.

**Comment by msimecek** [comment]
@SeanFeldman I can provide you with the tools, package manifest and commands which are needed to build an APPX package, so that you can build the process on AppVeyor. I currently don't have the capacity to learn how to migrate it. Email me at martin.simecek@microsoft.com, please.

After that someone needs to publish it to the Store using their developer account (ideally @paolosalvatori as the owner, but it can be anyone who is able to support it).

**Comment by SeanFeldman** [comment]
Sent an email. Thank you @msimecek

**Comment by SeanFeldman** [comment]
For signing: we could use https://letsencrypt.org/ to get a free certificate

**Comment by SeanFeldman** [comment]
With #88 completed, this is unblocked.

**Comment by SeanFeldman** [comment]
Update: i'll be done with conferences this week and plan to get back to this issue.

**Comment by tomkerkhove** [comment]
What's the status here? I think this can be closed now that we support Chocolatey?
https://chocolatey.org/packages/ServiceBusExplorer

**Comment by SeanFeldman** [comment]
Chocolatey is solely a _method_ of distribution. Desktop Bridge is more
than that.

I can't say if we should close it or leave one as we keep saying we'll have
a discussion we're this project is heading, and discussion is constantly
postponed.

Will defer to Paolo to make the call.

On Dec 20, 2017 12:59, "Tom Kerkhove" <notifications@github.com> wrote:

What's the status here? I think this can be closed now that we support
Chocolatey?
https://chocolatey.org/packages/ServiceBusExplorer

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub
<https://github.com/paolosalvatori/ServiceBusExplorer/issues/49#issuecomment-353137331>,
or mute the thread

**Comment by tomkerkhove** [comment]
What do you think @paolosalvatori?

**Comment by paolosalvatori** [comment]
@tomkerkhove we can try to convert the Service Bus Explorer to a Universal Windows Platform app and see if it works. I don't know how many potential users can be interested in running the tool as a UWP app, but we can definitely give it a try. @tomkerkhove, do you have the bandwith to give a try? I'll be very busy the next few weeks, and I won't have bandwidth till the end of this month. :(

**Comment by tomkerkhove** [comment]
Unfortunately I do not have the bandwidth and don't have any UI skills. That said, I'm not sure if UWP is the right fit as well given the multi-platform world Service Bus & we are in.

**Comment by paolosalvatori** [comment]
Exactly, I see any value in creating and maintaining a UWP version of the application. Chocolatey should suffice our needs.

**Comment by calloncampbell** [comment]
If you wanted to have this tool run on more than just Windows why not follow how the tool "Microsoft Azure Storage Explorer" was built with Electron. 

Otherwise, if you wanted to move it to UWP there was a lot of announcements at Build 2018 where UWP is now supported across all Windows frameworks: Winforms, WPF, and Win32.

**Comment by SeanFeldman** [comment]
> If you wanted to have this tool run on more than just Windows why not follow how the tool "Microsoft Azure Storage Explorer" was built with Electron.

@calloncampbell things are not as simple as "just do X and profit".
1. Historical aspect - the tool has been created using Windows Forms and switching to anything else is not something to easily ignore.
1. There's UI work that the current maintainers can or cannot do. Electron requires front-end developemnt that the current mainters don't do.
1. The current tool is using ASB .NET Client. Electron requires node.js library which is not on part with .NET client. While there's an option of going with .Electron.NET, there will be still some challenges associated with the new ASB C# client.

Saying that, the cross-platform question is something that we've raised and discussed.

**Comment by calloncampbell** [comment]
I wasn’t suggesting it would be as easy as “do X and profit”. I know it’s base don Winforms, I was just offering up some suggestions. If you want to have deployed via the Microsoft Store it’s definitely feasible. If however you wanted to go down the path of multi-platform then that is a much bigger task.

Callon

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

**Comment by SeanFeldman** [comment]
@MatthewSteeples that's great.
Feature branch might need to be synched with dev for other changes, but nothing's should be blocking you ATM.

**Comment by MatthewSteeples** [comment]
Looking at the recent tooling, the hardest part of this is going to be getting some logos! My artistic skills are next to zero, so if anyone has any ideas for images for the following sizes that'd be great! Need a square one that looks good at 300x300, and one for 88x88 (can be the same one if it works fine). Then there's also a banner size one 620x300.

**Comment by MatthewSteeples** [comment]
Found a bug already: You can't save connections as it tries saving to the install location rather than the user's profile. I'll have a look at seeing if that can be changed without breaking existing behaviour

**Comment by calloncampbell** [comment]
@MatthewSteeples I'm curious to the process you used for publishing it to the store. Can you provide details please. Is that tracked in a branch/PR I can reference?

**Comment by MatthewSteeples** [comment]
@calloncampbell The changes I've had to make are in a fork on [my own GitHub](https://github.com/MatthewSteeples/ServiceBusExplorer). The summary is that you create a "package project" in VS, and then when you build that it gives you an appx that you can submit to the store. It's a very manual process though, found [here](https://partner.microsoft.com/en-us/dashboard/windows/overview)

**Comment by calloncampbell** [comment]
Thanks @MatthewSteeples , I'll take a look.

**Comment by SeanFeldman** [comment]
Great work.
Once those steps can be automated, we could add it to our automated
deployment. Would require store account associated with Paolo.

**Comment by MatthewSteeples** [comment]
@SeanFeldman: @paolosalvatori will need to [create his own account](https://docs.microsoft.com/en-us/windows/uwp/publish/account-types-locations-and-fees) by the looks of things (if he creates a company account then it can have multiple users being members). There looks to be a cost for this of $99, and I can't find any exceptions listed for Open Source software.

Alternatively, we'd be happy to keep "hosting" it under ours. Failing that, from the latest version of Windows (1809) you can actually publish the packages on your own website (eg GitHub releases) and then they can be "installed" from there. They behave the same as store apps, and can self-update, but they bypass the store itself. It's basically an evolution of ClickOnce.

**Comment by SeanFeldman** [comment]
I'll let him decide on that. There's also individual account

> - Costs approximately $19 USD (one-time registration fee; the exact amount varies depending on your country or region)

with a perk

> - Shorter account verification process

**Comment by MatthewSteeples** [comment]
Just noticed where @paolosalvatori works. He might be able to get a discount ;)

**Comment by paolosalvatori** [comment]
Hi @MatthewSteeples @SeanFeldman sorry for coming late to the discussion, but i was extremely busy delivering design sessions on customer site in the last two weeks! Thanks a lot Matthew for the great work! I'll create an account on Microsoft store as soon as possible and I'll send it to you for the proper changes so that you move the location of the tool. Thanks :)

**Comment by paolosalvatori** [comment]
Hi @MatthewSteeples I just checked and found that I have an account already called "Paolo Salvatori". Please write me on paolos@microsoft.com and let's make the necessary changes together, thanks!

**Comment by paolosalvatori** [comment]
Hi @MatthewSteeples whenever you have time we should work together to move the tool under my account on the Windows store. Thanks for your collaboration!
